File: BackEnd/backend/src/main/java/com/fitapp/backend/routinecomplete/aplication/dto/routineexercise/request/ReorderSessionExercisesRequest.java
package com.fitapp.backend.routinecomplete.aplication.dto.routineexercise.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.List;

@Data
public class ReorderSessionExercisesRequest {

    @JsonProperty("dayOfWeek")
    private String dayOfWeek;

    @JsonProperty("sessionNumber")
    private Integer sessionNumber;

    @NotNull
    @NotEmpty
    @JsonProperty("exerciseIds")
    private List<Long> exerciseIds;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/feature/routine/model/rutinexercise/request/ReorderSessionExercisesRequest.kt
package com.fitapp.appfit.feature.routine.model.rutinexercise.request

import com.google.gson.annotations.SerializedName

data class ReorderSessionExercisesRequest(
    @SerializedName("dayOfWeek") val dayOfWeek: String? = null,
    @SerializedName("sessionNumber") val sessionNumber: Int? = null,
    @SerializedName("exerciseIds") val exerciseIds: List<Long>
)
